### PR TITLE
bpo-39372: Clean header files of declared interfaces with no implementations

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue39372

### CODE REVIEW

File: Include/methodobject.h [hunk 1]
```diff
@@ -22,8 +22,6 @@ typedef PyObject *(*PyCFunctionWithKeywords)(PyObject *, PyObject *,
 typedef PyObject *(*_PyCFunctionFastWithKeywords) (PyObject *,
                                                    PyObject *const *, Py_ssize_t,
                                                    PyObject *);
-typedef PyObject *(*PyNoArgsFunction)(PyObject *);
```
Comment (serhiy-storchaka) [inline Include/methodobject.h:25]: This **is** the definition. But it is not used, and never was used in CPython, and using it in extensions is likely an indication of an error.

File: Include/import.h [hunk 2]
```diff
@@ -81,8 +81,6 @@ PyAPI_FUNC(int) PyImport_ImportFrozenModule(
     const char *name            /* UTF-8 encoded string */
     );
 
-PyAPI_DATA(PyTypeObject) PyNullImporter_Type;
```
Comment (serhiy-storchaka) [inline Include/import.h:84]: It was implemented in 3.2. Removing the implementation in 3.3 was technically the breakage of the stable ABI. But is too late to fix this.

File: Include/bytesobject.h [hunk 3]
```diff
@@ -106,28 +106,6 @@ PyAPI_FUNC(int) PyBytes_AsStringAndSize(
                            strings) */
     );
 
-/* Using the current locale, insert the thousands grouping
-   into the string pointed to by buffer.  For the argument descriptions,
-   see Objects/stringlib/localeutil.h */
-#ifndef Py_LIMITED_API
-PyAPI_FUNC(Py_ssize_t) _PyBytes_InsertThousandsGroupingLocale(char *buffer,
```
Comment (serhiy-storchaka) [inline Include/bytesobject.h:113]: `_Py_InsertThousandsGrouping` in `Objects/stringlib` seems also not used.

File: Misc/NEWS.d/next/Core and Builtins/2020-01-17-17-51-05.bpo-39372.0IXPJZ.rst [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Clean header files of interfaces defined but with no implementation. The
```
Comment (serhiy-storchaka) [inline Misc/NEWS.d/next/Core and Builtins/2020-01-17-17-51-05.bpo-39372.0IXPJZ.rst]: It should be in the C API section.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM, except that the C API section is more appropriate for the NEWS entry.

File: Misc/NEWS.d/next/C API/2020-01-17-19-25-48.bpo-39372.hGJMY6.rst [hunk 5]
```diff
@@ -0,0 +1,9 @@
+Clean header files of interfaces defined but with no implementation. The
+public API symbols being removed are:
+
```
Comment (serhiy-storchaka) [inline Misc/NEWS.d/next/C API/2020-01-17-19-25-48.bpo-39372.hGJMY6.rst]: I afraid that using multi-paragraph news entry can cause problems when files be merged into a single NEWS file.

Comment (pablogsal) [reply Misc/NEWS.d/next/C API/2020-01-17-19-25-48.bpo-39372.hGJMY6.rst]: Thanks for pointing that out! ... it was too good to be true (I couldn't check at the time I pushed this change).

**Comment by pablogsal** [comment]
Thanks, Serhiy for the review!

**Comment by serhiy-storchaka** [comment]
Thank you for your great work!

**Pull request merged**